src/units/abilities.cpp

```diff
@@ -429,19 +411,47 @@ unit::recursion_guard& unit::recursion_guard::operator=(unit::recursion_guard&&
 unit::recursion_guard::~recursion_guard()
 {
 	if(parent) {
-		assert(parent->num_recursion_ > 0);
-		parent->num_recursion_--;
+		assert(!parent->open_queries_.empty());
+		parent->open_queries_.pop_back();
+	}
+}
+
+bool unit::get_self_ability_bool_impl(const config& ability, const std::string& tag_name, const map_location& loc) const
+{
+	unit::recursion_guard filter_lock;
+	filter_lock = update_variables_recursion(ability);
```

Comment:
When the code is like this, with no conditional block around the `update_variables_recursion` call, it doesn't need a separate line for declaring `unit::recursion_guard filter_lock;`. Replace both lines with
```
auto filter_lock = update_variables_recursion(ability);
```

Reply:
done